rskj-core/src/main/java/co/rsk/mine/gas/provider/StableMinGasPriceProvider.java

```diff
@@ -63,17 +101,41 @@ private long calculateMinGasPriceBasedOnBtcPrice(long btcValue) {
         return minStableGasPrice / btcValue;
     }
 
-    @Override
-    public Coin getMinGasPriceAsCoin() {
-        return Coin.valueOf(getMinGasPrice());
+    private synchronized Future<Long> fetchPriceAsync() {
+        Future<Long> curFuture = priceFuture.get();
+        if (curFuture != null) {
+            return curFuture;
+        }
+
+        CompletableFuture<Long> newFuture = new CompletableFuture<>();
+        priceFuture.set(newFuture);
+
+        new Thread(() -> {
+            Optional<Long> priceResponse = fetchPriceSync();
+            newFuture.complete(priceResponse.orElse(getLastMinGasPrice()));
+            priceFuture.set(null);
+        }).start();
+
+        return newFuture;
     }
 
-    private void fetchPrice() {
+    private Optional<Long> fetchPriceSync() {
         Optional<Long> priceResponse = getBtcExchangeRate();
         if (priceResponse.isPresent() && priceResponse.get() > 0) {
-            lastMinGasPrice = calculateMinGasPriceBasedOnBtcPrice(priceResponse.get());
+            long result = calculateMinGasPriceBasedOnBtcPrice(priceResponse.get());
+            lastMinGasPrice = result;
+            lastUpdateTimeMillis = System.currentTimeMillis();
+            numOfFailures.set(0);
+            return Optional.of(result);
+        }
+
+        int curNumOfFailures = numOfFailures.incrementAndGet();
```

Comment:
nitpick:

Also here, maybe `currentFailureCounter` or `currentNumberOfFailures` would be fine. IMHO, no there is no much need for these small abbreviations.